jbpm-audit/src/main/java/org/jbpm/process/audit/event/DefaultAuditEventBuilderImpl.java

```diff
@@ -55,8 +55,13 @@ public AuditEvent buildEvent(ProcessStartedEvent pse) {
         	log.setCorrelationKey(correlationKey.toExternalForm());
         }
         try {
-            long parentProcessInstanceId = (Long) pi.getMetaData().get("ParentProcessInstanceId");
-            log.setParentProcessInstanceId(parentProcessInstanceId);
+            Object value = pi.getMetaData().get("ParentProcessInstanceId");
+	    if( value != null ){
+	        long parentProcessInstanceId = (Long) value;
+                log.setParentProcessInstanceId( parentProcessInstanceId );
+	     }else{
+		log.setParentProcessInstanceId(-1L);
+	     }
```

Comment:
wouldn't this be way easier to use a single line as follows instead of all these checks...

```
long parentProcessInstanceId = (Long) pi.getMetaData().getOrDefault("ParentProcessInstanceId", -1L);
```

Reply:
Sounds perfect to me. I will apply it. Let me check!.